test/src/main/java/org/apache/accumulo/test/ExternalCompactionIT.java

```diff
@@ -193,22 +195,55 @@ public boolean accept(Key k, Value v) {
 
   }
 
+  @Before
+  public void setUp() throws Exception {
+    if (SharedMiniClusterBase.getCluster() == null) {
+      SharedMiniClusterBase.startMiniClusterWithConfig(this);
+    }
+  }
+
+  private void stopProcess(Process process) throws Exception {
+    if (process.supportsNormalTermination()) {
+      process.destroyForcibly();
+    } else {
+      LOG.info("Stopping process manually");
+      new ProcessBuilder("kill", Long.toString(process.pid())).start();
+      process.waitFor();
+    }
+  }
+
+  private void cleanupTables(String... tables) {
+    try (AccumuloClient client = Accumulo.newClient()
+        .from(SharedMiniClusterBase.getCluster().getClientProperties()).build()) {
+      for (String table : tables) {
+        try {
+          client.tableOperations().delete(table);
+        } catch (Exception e) {
+          fail("Error deleting table: " + table + ", msg: " + e.getMessage());
+        }
+      }
+    }
+  }
```

Comment:
Looks like you have a unique table name per test. If that is the case do you need to delete them after each test?

Reply:
I do need to delete them, or maybe change some other code. There are places in the tests where I am checking the external compaction final state markers in the metadata table and I'm expecting one entry. If I don't delete the table, then I end up with more than one. IIRC there is not a tableId in the final state marker for me to use to verify that it's complete. I'll double-check.

Reply:
Should/can the tableId be added to the final state marker?

Reply:
So, I just spent hours trying to figure out why a test was failing to complete after switching the IT to use a different parent class that does not restart MAC between tests. The test was running a compaction from the previous test due to the way the previous test ended. When the compactor and coordinator started for the current test the tablet server gave it a compaction from the prior test. I guess I should be happy that it's resilient to components being restarted, but I was banging my head against the wall. For the purposes of the IT, I have to delete the tables between tests.

Reply:
Addressed in da6017f by deleting tables in tearDown() after each test.